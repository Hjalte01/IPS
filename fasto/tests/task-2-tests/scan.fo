fun [char] write_str([char] x) = write(x)
fun [[char]] write_str_arr([[char]] xs) = map(write_str, xs)

fun char write_char(char x) = write(x)
fun [char] write_char_arr([char] xs) = map(write_char, xs) 

fun bool write_bool(bool x) = write(x)
fun [bool] write_bool_arr([bool] xs) = map(write_bool, xs)

fun int write_int(int x) = write(x)
fun [int] write_int_arr([int] xs) = map(write_int, xs)


fun int incr(int a, int b) = a + b
fun int decr(int a, int b) = a - b
fun bool ander(bool a, bool b) = a && b
fun char first_char(char a, char b) = a
fun char second_char(char a, char b) = b
fun [char] first_str([char] a, [char] b) = a
fun [char] second_str([char] a, [char] b) = b


fun int main() =
  let a = write_int_arr(scan(incr, 0, {1, 2, 3})) in
  let b = write_int_arr(scan(decr, 0, {1, 2, 3})) in
  let c = write_bool_arr(scan(ander, true, {true, false, true})) in
  let d = write_char_arr(scan(first_char, 'a', {'a', 'b', 'c'})) in
  let e = write_char_arr(scan(second_char, 'a', {'a', 'b', 'c'})) in
  let g = write_str_arr(scan(first_str, "good ", {"life", "day", "time"})) in
  let h = write_str_arr(scan(second_str, "good ", {"life", "day", "time"})) in
  0


