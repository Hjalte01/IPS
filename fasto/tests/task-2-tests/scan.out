1 3 6 
-1 -3 -6 
true false false 
a a a 
a b c 
good good good 
life day time 
